server/service/src/stocktake/validate.rs

```diff
@@ -18,10 +18,12 @@ pub fn check_stocktake_not_finalised(status: &StocktakeStatus) -> bool {
 pub fn check_no_stocktake_lines_exist(
     connection: &StorageConnection,
     stocktake_line_id: &str,
+    store_id: &str,
 ) -> Result<bool, RepositoryError> {
-    let result = StocktakeLineRepository::new(connection).count(Some(
-        StocktakeLineFilter::new().stocktake_id(EqualFilter::equal_to(stocktake_line_id)),
-    ))?;
+    let result = StocktakeLineRepository::new(connection).count(
+        Some(StocktakeLineFilter::new().stocktake_id(EqualFilter::equal_to(stocktake_line_id))),
+        Some(store_id.to_string()),
```

Comment:
Is is safer to have this a `None` just incase the same id could exist at a different store?

Reply:
the store_id is only for use in the item code or name filter, which isn't used in this call - so passing in store_id or None would have the same effect.. though could be confusing to people reading the code